CMS-Enterprise/mint-app
# [MINT-3182] MTO Common Milestones and Common Solutions

# MINT-3182

## Description

Steven's notes
1. Created Common Milestone, Common Solution, and Common Milestone Solution Link table
2. Created a few test entries in the database
3. Data loaders for Common Solutions and Milestones
   a. Currently, it will provide contextual data about if a model is using a common milestone or common solution  ONLY at the top level.
   b. EX if you look for a solution associated with a common milestone, it will not provide any meta data. We can expand later if this is needed
 4. Common Solution says that the solution is added ONLY for the Model Plan. It doesn't return data about if it was added for the Common Milestone. 


## How to test this change

Steven's instructions
1.  Use the new postman queries
2. Query common Milestones
   a. At the Model To Operation Matrix top level
       i. Add the milestone and query again, note that the solution is now marked added
   b. At the top level query by key
       i. note is added is always false
3. Query Common Solutions
       a. At the Model To Operation Matrix top level
           i. Is added is based on if the Solution is added to a Model Plan)
       b. At the top level query by key
             i. note is added is always false
         c. At the Common Milestone Level
             i. note is added is always false (We could expand this if desired, but would need to interpret if we should check if it's added to a model plan, or that specifc need type.). This could create issues if the field means different things at different times.
 

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
